src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+    public static String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate toDate = LocalDate.parse(dateTo, formatter);
+
+        Map<String, Integer> earnings = new HashMap<>();
+        for (String name : names) {
+            earnings.put(name, 0);
+        }
+
+        for (String line : data) {
+            String[] parts = line.split(" ");
+            LocalDate date = LocalDate.parse(parts[0], formatter);
```

Comment:
Please, create constants for all numbers and strings like this `" "`

Reply:
not fixed

Reply:
array indexes not fixed, they should also be extracted and have informative names